azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusExtractor.java

```diff
@@ -180,7 +180,7 @@ private void extractConditions() {
     Map<String, V1PodCondition> conditionMap = new HashMap<>();
     this.podConditions.stream().forEach(
         condition ->
-            conditionMap.put(condition.getType(), condition));
+            conditionMap.put(String.valueOf(condition.getType()), condition));
```

Comment:
Instead of converting to string, use the TypeEnum as key instead of String. This is likely more efficient and saves the conversion both here and at line 228.

Reply:
I've changed it to .getname() which is essentially pulling it from an Enum type.

Reply:
Changing the type of the entire map, could be part of another change ? Since it's not relevant to this upgrade.